src/liftdrag_plugin/advanced_liftdrag_plugin.cpp

```diff
@@ -0,0 +1,602 @@
+      const auto lift_force_topic = this->sdf->Get<std::string>("topic_name");
+      lift_force_pub_ = node_handle_->Advertise<physics_msgs::msgs::Force>("~/" + lift_force_topic);
+      gzdbg << "Publishing to ~/" << lift_force_topic << std::endl;
+  }
+
+  if (_sdf->HasElement("windSubTopic")){
+    this->wind_sub_topic_ = _sdf->Get<std::string>("windSubTopic");
+    wind_sub_ = node_handle_->Subscribe("~/" + wind_sub_topic_, &AdvancedLiftDragPlugin::WindVelocityCallback, this);
+  }
+
+}
+
+/////////////////////////////////////////////////
+void AdvancedLiftDragPlugin::OnUpdate()
+{
+  GZ_ASSERT(this->link, "Link was NULL");
+  // get linear velocity at ref_pt in inertial frame
+#if GAZEBO_MAJOR_VERSION >= 9
+  ignition::math::Vector3d air_velocity = this->link->WorldLinearVel(this->ref_pt) - wind_vel_;
+  const common::Time current_time = this->world->SimTime();
+#else
+  ignition::math::Vector3d air_velocity = ignitionFromGazeboMath(this->link->GetWorldLinearVel(this->ref_pt)) - wind_vel_;
+  const common::Time current_time = this->world->GetSimTime();
+#endif
+  const double dt = (current_time - this->last_pub_time).Double();
+  // pose of body
+#if GAZEBO_MAJOR_VERSION >= 9
+  ignition::math::Pose3d pose = this->link->WorldPose();
+#else
+  ignition::math::Pose3d pose = ignitionFromGazeboMath(this->link->GetWorldPose());
+#endif
+
+  //Define body frame: X forward, Z downward, Y out the right wing
+  ignition::math::Vector3d body_x_axis = pose.Rot().RotateVector(this->forward);
+  ignition::math::Vector3d body_z_axis = -1*(pose.Rot().RotateVector(this->upward));
+  ignition::math::Vector3d body_y_axis =  body_z_axis.Cross(body_x_axis);
+
+  // Get the in-plane velocity (remove spanwise velocity from the velocity vector air_velocity)
+  ignition::math::Vector3d velInLDPlane = air_velocity - air_velocity.Dot(body_y_axis)*body_y_axis;
+  double speedInLDPlane = velInLDPlane.Length();
+
+  // Define stability frame: X is in-plane velocity, Y is the same as body Y,
+  // and Z perpendicular to both
+  ignition::math::Vector3d stability_x_axis = velInLDPlane/speedInLDPlane;
+  ignition::math::Vector3d stability_y_axis = body_y_axis;
+  ignition::math::Vector3d stability_z_axis = stability_x_axis.Cross(stability_y_axis);
+
+  double span = std::sqrt(this->area*this->AR); // Wing span
+  if (this->mac == 0){
+    //Computing the mean aerodynamic chord involves integrating the square of
+    //the chord along the span. If this parameter has not been input, this plugin
+    //will approximate MAC as mean chord. This works for rectangular and trapezoidal
+    //wings, but for more complex wing shapes, doing the integral is preferred.
+    this->mac = this->area/span;
+  }
+
+  //Get non-dimensional body rates. Gazebo uses ENU, so some have to be flipped
+  ignition::math::Vector3d body_rates = this->link->RelativeAngularVel();
+  double airspeed = air_velocity.Length();
+  double p = body_rates.X()*span/(2*airspeed); //Non-dimensionalized roll rate
```

Comment:
I see you took out the returns according to these to comments:
https://github.com/PX4/PX4-SITL_gazebo/pull/901#discussion_r964806382
https://github.com/PX4/PX4-SITL_gazebo/pull/901#discussion_r964810851

however, the latter part on multiplying the quadratic airspeed through the full equation so that the denominator goes away in these terms was not done.

so currently if airspeed is zero, it'll blow up.. right?

Reply:
@tstastny Thanks for the suggestion: I think I got it with the latest commit.